app/upgrades/v9/upgrades.go

```diff
@@ -0,0 +1,64 @@
+package v9
+
+import (
+	"fmt"
+
+	errorsmod "cosmossdk.io/errors"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/module"
+	upgradetypes "github.com/cosmos/cosmos-sdk/x/upgrade/types"
+
+	claimkeeper "github.com/Stride-Labs/stride/v8/x/claim/keeper"
+)
+
+// CreateUpgradeHandler creates an SDK upgrade handler for v29
+func CreateUpgradeHandler(
+	mm *module.Manager,
+	configurator module.Configurator,
+	claimKeeper claimkeeper.Keeper,
+) upgradetypes.UpgradeHandler {
+	return func(ctx sdk.Context, _ upgradetypes.Plan, vm module.VersionMap) (module.VersionMap, error) {
+		ctx.Logger().Info("Starting upgrade v9...")
+
+		if err := AddFieldsToAirdropType(ctx, claimKeeper); err != nil {
+			return vm, errorsmod.Wrapf(err, "unable to update airdrop schema")
+		}
+
+		ctx.Logger().Info("Running module migrations...")
+		return mm.RunMigrations(ctx, configurator, vm)
+	}
+}
+
+func AddFieldsToAirdropType(ctx sdk.Context, claimKeeper claimkeeper.Keeper) error {
+	ctx.Logger().Info("Adding additional fields to airdrop struct...")
+
+	// Get list of airdrops from claim parameters
+	claimParams, err := claimKeeper.GetParams(ctx)
+	if err != nil {
+		return errorsmod.Wrapf(err, "unable to get claim parameters")
+	}
+
+	for _, airdrop := range claimParams.Airdrops {
+		// Add the chain ID to each airdrop
+		chainId, ok := AirdropChainIds[airdrop.AirdropIdentifier]
+		if !ok {
+			ctx.Logger().Error(fmt.Sprintf("Chain ID not specified for %s airdrop", chainId))
```

Comment:
nit: should we assign `airdrop.ChainId = ""` here?

Reply:
hmm good question

Reply:
ah actually, that should happen by default

Reply:
I think the questions is more should we leave it as an empty string and skip over or throw an error here.
I think I lean towards empty string so we don't halt the chain

Reply:
just added a test case for the empty chain ID

Reply:
> I lean towards empty string so we don't halt the chain
Agreed

Reply:
All of our current airdrops should map onto chains, so I think we should throw an error here (I don't see why it would throw an error, but if we can't find a chainId for an airdrop maybe something has gone wrong)

Reply:
chatted offline, leaving it be considering how low the impact would be if this situation happened